### PR TITLE
Improve Furigana Regex to Target Kanji Characters More Accurately

### LINKED ISSUE #2822: Furigana stretches across kana characters
Note: I created this issue to link it in my pull request

**Problem:**
When using the {{furigana:fiel}} syntax, the rendered furigana stretches across other kana characters, not just kanji. This results in the furigana being hard to read:

Screenshot from 2023-11-11 10-15-47
Screenshot from 2023-11-11 10-15-30

**Expected behaviour:**

Furigana is only displayed above kanji characters:

image

### PR DESCRIPTION
**Pull request by AlexW00**
TLDR: fixes #2822 by adjusting the FURIGANA regex to target kanji characters more accurately.

## Description
This pull request updates the regular expression used in the `furigana_filter` function to better target kanji characters for furigana. The previous regex pattern was not specific to kanji and could erroneously apply furigana to sequences of characters that included hiragana or were solely hiragana. The updated regex is designed to more accurately identify kanji characters, thereby improving the accuracy of furigana application.

## Changes
The key change is in the `FURIGANA` regex within the `furigana_filter` function:

```rust
lazy_static! {
    static ref FURIGANA: Regex = Regex::new(r" ?([\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF]+)\[(.+?)\]").unwrap();
}
```

This new pattern includes the following Unicode ranges:
- `\u3400-\u4DBF`: CJK Unified Ideographs Extension A
- `\u4E00-\u9FFF`: CJK Unified Ideographs
- `\uF900-\uFAFF`: CJK Compatibility Ideographs

These ranges are more representative of kanji characters. The regex now matches sequences of these characters followed by square brackets containing furigana.

## Showcase

Example field:

Screenshot from 2023-11-11 10-15-47

### Currently:

`{{furigana:japanese}}` renders as:

Screenshot from 2023-11-11 10-15-30

### With PR:

`{{furigana:japanese}}` renders as:

Screenshot from 2023-11-11 10-20-30

### CODE REVIEW

**Comment by dae** [comment]
I'm afraid I'm not keen on this change, as the issue lies in your input. The furigana feature expects spaces to be used to delineate the kanji boundaries. If you use an add-on like https://ankiweb.net/shared/info/3918629684 to generate the readings instead of writing them manually, it will automatically do this for you.

**Comment by dae** [comment]
(this is also documented in the manual: https://docs.ankiweb.net/templates/fields.html#ruby-characters)
